libs/librepcbcommon/version.cpp

```diff
@@ -127,38 +151,56 @@ Version& Version::operator=(const Version& rhs) noexcept
 
 bool Version::operator>(const Version& rhs) const noexcept
 {
-    if (mNumbers.isEmpty()) return false;
-    return (compare(rhs) > 0);
+    if (isValid() && rhs.isValid()) {
+        return (compare(rhs) > 0);
+    } else {
+        return false;
+    }
 }
 
 bool Version::operator<(const Version& rhs) const  noexcept
 {
-    if (mNumbers.isEmpty()) return false;
-    return (compare(rhs) < 0);
+    if (isValid() && rhs.isValid()) {
+        return (compare(rhs) < 0);
+    } else {
+        return false;
+    }
 }
 
 bool Version::operator>=(const Version& rhs) const noexcept
 {
-    if (mNumbers.isEmpty()) return false;
-    return (compare(rhs) >= 0);
+    if (isValid() && rhs.isValid()) {
+        return (compare(rhs) >= 0);
+    } else {
+        return false;
+    }
 }
 
 bool Version::operator<=(const Version& rhs) const noexcept
 {
-    if (mNumbers.isEmpty()) return false;
-    return (compare(rhs) <= 0);
+    if (isValid() && rhs.isValid()) {
+        return (compare(rhs) <= 0);
+    } else {
+        return false;
+    }
 }
 
 bool Version::operator==(const Version& rhs) const noexcept
 {
-    if (mNumbers.isEmpty()) return false;
-    return (compare(rhs) == 0);
+    if (isValid() && rhs.isValid()) {
+        return (compare(rhs) == 0);
+    } else {
+        return false;
+    }
 }
 
 bool Version::operator!=(const Version& rhs) const noexcept
 {
-    if (mNumbers.isEmpty()) return false;
-    return (compare(rhs) != 0);
+    if (isValid() && rhs.isValid()) {
+        return (compare(rhs) != 0);
+    } else {
+        return false;
+    }
```

Comment:
This doesn't make any sense semantically. If any of the values is invalid (`isValid()` returns false) the `!=` will return true which means the objects are equal!

I think it should behave the same as a float with NaN.